anvill/python/anvill/arch.py

```diff
@@ -1005,127 +1066,125 @@ class AArch32Arch(Arch):
+        "D0": _REG_FAMILY_Q0,
+        "D1": _REG_FAMILY_Q0,
+        "D2": _REG_FAMILY_Q1,
+        "D3": _REG_FAMILY_Q1,
+        "D4": _REG_FAMILY_Q2,
+        "D5": _REG_FAMILY_Q2,
+        "D6": _REG_FAMILY_Q3,
+        "D7": _REG_FAMILY_Q3,
+        "D8": _REG_FAMILY_Q4,
+        "D9": _REG_FAMILY_Q4,
+        "D10": _REG_FAMILY_Q5,
+        "D11": _REG_FAMILY_Q5,
+        "D12": _REG_FAMILY_Q6,
+        "D13": _REG_FAMILY_Q6,
+        "D14": _REG_FAMILY_Q7,
+        "D15": _REG_FAMILY_Q7,
+        "D16": _REG_FAMILY_Q8,
+        "D17": _REG_FAMILY_Q8,
+        "D18": _REG_FAMILY_Q9,
+        "D19": _REG_FAMILY_Q9,
+        "D20": _REG_FAMILY_Q10,
+        "D21": _REG_FAMILY_Q10,
+        "D22": _REG_FAMILY_Q11,
+        "D23": _REG_FAMILY_Q11,
+        "D24": _REG_FAMILY_Q12,
+        "D25": _REG_FAMILY_Q12,
+        "D26": _REG_FAMILY_Q13,
+        "D27": _REG_FAMILY_Q13,
+        "D28": _REG_FAMILY_Q14,
+        "D29": _REG_FAMILY_Q14,
+        "D30": _REG_FAMILY_Q15,
+        "D31": _REG_FAMILY_Q15,
+        "S0": _REG_FAMILY_Q0,
+        "S1": _REG_FAMILY_Q0,
+        "S2": _REG_FAMILY_Q0,
+        "S3": _REG_FAMILY_Q0,
+        "S4": _REG_FAMILY_Q1,
+        "S5": _REG_FAMILY_Q1,
+        "S6": _REG_FAMILY_Q1,
+        "S7": _REG_FAMILY_Q1,
+        "S8": _REG_FAMILY_Q2,
+        "S9": _REG_FAMILY_Q2,
+        "S10": _REG_FAMILY_Q2,
+        "S11": _REG_FAMILY_Q2,
+        "S12": _REG_FAMILY_Q3,
+        "S13": _REG_FAMILY_Q3,
+        "S14": _REG_FAMILY_Q3,
+        "S15": _REG_FAMILY_Q3,
+        "S16": _REG_FAMILY_Q4,
+        "S17": _REG_FAMILY_Q4,
+        "S18": _REG_FAMILY_Q4,
+        "S19": _REG_FAMILY_Q4,
+        "S20": _REG_FAMILY_Q5,
+        "S21": _REG_FAMILY_Q5,
+        "S22": _REG_FAMILY_Q5,
+        "S23": _REG_FAMILY_Q5,
+        "S24": _REG_FAMILY_Q6,
+        "S25": _REG_FAMILY_Q6,
+        "S26": _REG_FAMILY_Q6,
+        "S27": _REG_FAMILY_Q6,
+        "S28": _REG_FAMILY_Q7,
+        "S29": _REG_FAMILY_Q7,
+        "S30": _REG_FAMILY_Q7,
+        "S31": _REG_FAMILY_Q7,
     }
 
-    def name(self):
+    def name(self) -> ArchName:
         return "aarch32"
 
-    def program_counter_name(self):
+    def program_counter_name(self) -> Register:
         return "PC"
 
-    def stack_pointer_name(self):
+    def stack_pointer_name(self) -> Register:
         return "SP"
 
-    def return_address_proto(self):
+    def return_address_proto(self) -> Dict:
         return {"register": "LR", "type": "I"}
 
-    def return_stack_pointer_proto(self, num_bytes_popped):
+    def return_stack_pointer_proto(self, num_bytes_popped) -> Dict:
         return {"register": "SP", "offset": abs(num_bytes_popped), "type": "I"}
 
-    def pointer_size(self):
+    def pointer_size(self) -> int:
         return 4
 
-    def register_family(self, reg_name):
+    def stack_offset(self) -> int:
+        return 0
+
+    def register_family(self, reg_name) -> Tuple[RegisterBounds, ...]:
         if reg_name.startswith("%"):
```

Comment:
Here we should do: `self._REG_FAMILY[self.register_name(reg_name)]`, and/or have the `reg_name` parameter of `register_family` be a `Register`.

Reply:
Thanks for catching this. It's a copy-paste issue from the `register_name`.

Reply:
Thanks for catching this. `register_family` implementation had redundant code. I updated it to use `register_name` directly. I think let the parameter as string type for the user and get `register_name` internally.